core-bundle/assets/styles/clipboard.pcss

```diff
@@ -0,0 +1,69 @@
+/* Button in listing row */
+#tl_listing button.clipboard {
+    width: 16px;
+    height: 16px;
+    vertical-align: inherit;
+    display: inline-flex;
+
+    .url {
+        display: none;
+    }
+}
+
+/* Button as part of an info message */
+.tl_message button.clipboard {
+    display: inline-flex;
+    align-items: center;
+    gap: 2px;
+
+    .url {
+        text-decoration: underline;
+        white-space: nowrap;
+        text-overflow: ellipsis;
+        max-width: 50vw;
+        overflow: hidden;
+    }
+}
+
+button.clipboard {
+    background-color: transparent;
+    border-radius: 3px;
+    border: none;
+    outline: none;
+    box-sizing: content-box;
+    padding: 0;
+
+    &--expired {
+        opacity: .5;
+    }
+
+    &:before {
+        flex-shrink: 0;
+        content: '';
+        width: 16px;
+        height: 16px;
+        background: url("data:image/svg+xml,%3Csvg fill='none' stroke='%23666' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' stroke-width='2' d='M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2'%3E%3C/path%3E%3C/svg%3E") no-repeat center center;
```

Comment:
I would prefer to have icon files since they are easier to reuse if we need a clipboard icon somewhere else. Also would be consistent with all the other CSS (apart from this being a separate file, not sure we want to go this route).

Reply:
Using files has some drawbacks: If the file wasn't requested before (e.g. regular page load), it would only be loaded once the button was clicked resulting in a flaky animation. How would you solve this?

Reply:
I would solve it with

```html
<link rel="preload" href="foobar.svg" as="image" type="image/svg+xml">
```